src/staking/Staking.ts

```diff
@@ -0,0 +1,54 @@
+import Web3 from 'web3'
+
+export default abstract class Staking {
```

Comment:
what you think about some renaming here?
the class name is confusing

I suggest `RewardProgram`

Reply:
okay i have changed the name to RewardProgram